app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt

```diff
@@ -182,6 +204,92 @@ class ProfileFragment : Fragment(), BottomIconDoubleClick {
         rootView.changePassword.setOnClickListener {
             handleChangePassword()
         }
+
+        rootView.deleteAccount.setOnClickListener {
+            openDeleteAccountDialog()
+        }
+    }
+
+    private fun openDeleteAccountFailDialog() {
+        AlertDialog.Builder(requireContext())
+            .setTitle(getString(R.string.delete_account))
+            .setMessage(getString(R.string.delete_account_fail_message))
+            .setPositiveButton(getString(R.string.ok)) { dialog, _ ->
+                dialog.cancel()
+            }.create()
+            .show()
+    }
+
+    private fun openDeleteAccountDialog() {
+        val deleteLayout = layoutInflater.inflate(R.layout.dialog_delete_account, null)
+        deleteLayout.confirmEmailLayout.isVisible = true
+
+        val dialog = AlertDialog.Builder(requireContext())
+            .setTitle(getString(R.string.delete_account))
+            .setView(deleteLayout)
+            .create()
+
+        deleteLayout.cancelButton.setOnClickListener {
+            dialog.cancel()
+        }
+
+        deleteLayout.confirmEmailButton.setOnClickListener {
+            deleteLayout.confirmDeleteLayout.isVisible = true
+            deleteLayout.confirmEmailLayout.isVisible = false
+        }
+
+        deleteLayout.deleteButton.setOnClickListener {
+            dialog.cancel()
+            confirmDeleteAccount()
+        }
+
+        deleteLayout.confirmDeleteCheckbox.setOnCheckedChangeListener { _, isChecked ->
+            deleteLayout.deleteButton.isEnabled = isChecked
+        }
+
+        deleteLayout.email.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
+                deleteLayout.confirmEmailButton.isEnabled = user?.email == s.toString()
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) { /* Do Nothing */ }
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) { /* Do Nothing */ }
+        })
+
+        dialog.show()
+    }
+
+    private fun confirmDeleteAccount() {
+        val progressDialog = progressDialog(context, getString(R.string.deleting_account))
+
+        loginViewModel.progress
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
+                progressDialog.show(it)
+            })
+
+        loginViewModel.validPassword
```

Comment:
Can you use chaining here, after password validation, directly use the data?

Reply:
I don't think so, chaining is more about using result from the previous request for the second request, should not be nested